pkg/plugins/policies/meshtimeout/api/v1alpha1/validator.go

```diff
@@ -12,7 +12,7 @@ import (
 func (r *MeshTimeoutResource) validate() error {
 	var verr validators.ValidationError
 	path := validators.RootedAt("spec")
-	verr.AddErrorAt(path.Field("targetRef"), r.validateTop(r.Spec.TargetRef))
+	verr.AddErrorAt(path.Field("targetRef"), r.validateTop(r.Spec.TargetRef, len(r.Spec.From) > 0 || len(r.Spec.Rules) > 0))
```

Comment:
maybe you should create a func
```go
func AffectsInbounds(p core_model.Policy) bool
```
that tries to cast to `PolicyWithRules` and `PolicyWithFromList` and checks `len`.

Otherwise we have to update these validations every time we add `spec.rules` support

Reply:
changed